ex4/ex4.c

```diff
@@ -10,6 +10,19 @@
 int main(int argc, char* argv[])
 {
     // Your code here    
-
+int rc = fork();
+    if (rc < 0) {
+        fprintf(stderr, "fork failed\n");
+        exit(1);
+    }
+    else if (rc == 0) {
+        printf("hello, child here (pid: %d) \n", (int) getpid());
+        char *args[]={"/bin/ls",NULL};
+        execvp(args[0], args);
+        printf("invisible");
+    } else {
+        int wc = waitpid(rc, NULL, 0);
```

Comment:
Another implicit declaration warning here on compile, I left some notes on the subject a bit earlier in ex3.